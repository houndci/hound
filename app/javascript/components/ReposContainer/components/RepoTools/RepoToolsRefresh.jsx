import React from 'react';

export default class RepoToolsRefresh extends React.Component {
  buttonText(isSyncing) {
    if (isSyncing) {
      return Hound.settings.syncingButtonText;
    } else {
      return Hound.settings.syncNowButtonText;
    }
  }

  render() {
    const { isSyncing, onRefreshClicked } = this.props;

    return (
      <div className="repo-tools-refresh">
        <button
          className="repo-tools-refresh-button"
          disabled={isSyncing ? "disabled" : null}
          onClick={onRefreshClicked}
        >
          <span>{this.buttonText(isSyncing)}</span>
        </button>
      </div>
    );
  }
}
