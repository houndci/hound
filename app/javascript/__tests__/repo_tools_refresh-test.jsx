import RepoToolsRefresh from '../components/ReposContainer/components/RepoTools/RepoToolsRefresh';

it('renders appropriately', () => {
  const onRefreshClicked = jest.fn();

  const wrapper = shallow(
    <RepoToolsRefresh
      isSyncing={false}
      onRefreshClicked={onRefreshClicked}
    />
  );
  expect(wrapper).toMatchSnapshot();
});

it('renders appropriately (when syncing)', () => {
  const onRefreshClicked = jest.fn();

  const wrapper = shallow(
    <RepoToolsRefresh
      isSyncing={true}
      onRefreshClicked={onRefreshClicked}
    />
  );
  expect(wrapper).toMatchSnapshot();
});
