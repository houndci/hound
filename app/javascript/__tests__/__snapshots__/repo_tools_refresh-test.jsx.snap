// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders appropriately (when syncing) 1`] = `
<div
  className="repo-tools-refresh"
>
  <button
    className="repo-tools-refresh-button"
    disabled="disabled"
    onClick={[MockFunction]}
  >
    <span />
  </button>
</div>
`;

exports[`renders appropriately 1`] = `
<div
  className="repo-tools-refresh"
>
  <button
    className="repo-tools-refresh-button"
    disabled={null}
    onClick={[MockFunction]}
  >
    <span />
  </button>
</div>
`;
